mm2src/mm2_main/src/lp_ordermatch/orderbook_events.rs

```diff
@@ -14,10 +14,13 @@ pub struct OrderbookStreamer {
     rel: String,
 }
 
-impl OrderbookStreamer {
-    pub fn new(ctx: MmArc, base: String, rel: String) -> Self { Self { ctx, base, rel } }
+impl<'a> DeriveStreamerId<'a> for OrderbookStreamer {
+    type InitParam = (MmArc, String, String);
+    type DeriveParam = (&'a str, &'a str);
```

Comment:
Perhaps something like this would help more on read:

```suggestion
type BaseAndRel = (&'a str, &'a str);
impl<'a> DeriveStreamerId<'a> for OrderbookStreamer {
    type InitParam = (MmArc, String, String);
    type DeriveParam = BaseAndRel;
```

Reply:
updated in 4b65bf6d6b1c68a5b0cad84d719745f1a19cb73c